doc/source/data/transforming-datasets.rst

```diff
@@ -4,64 +4,85 @@
+     - Apply a given function to batches of records of this dataset. 
+   * - :meth:`ds.split() <ray.data.Dataset.split>`
+     - One-to-one
+     - | Split the dataset into N disjoint pieces.
+   * - :meth:`ds.repartition(shuffle=False) <ray.data.Dataset.repartition>`
+     - One-to-one
+     - | Repartition the dataset into N blocks, without shuffling the data.
+   * - :meth:`ds.repartition(shuffle=True) <ray.data.Dataset.repartition>`
+     - All-to-all
+     - | Repartition the dataset into N blocks, shuffling the data during repartition.
+   * - :meth:`ds.random_shuffle() <ray.data.Dataset.random_shuffle>`
+     - All-to-all
+     - | Randomly shuffle the elements of this dataset.
+   * -  :meth:`ds.sort() <ray.data.Dataset.sort>`
+     - All-to-all
+     - | Sort the dataset by a sortkey.
+   * -  :meth:`ds.groupby() <ray.data.Dataset.groupby>`
+     - All-to-all
+     - | Group the dataset by a groupkey.
 
 .. tip::
 
-    Datasets also provides the convenience methods ``map``, ``flat_map``, and ``filter``, which are not vectorized (slower than ``map_batches``), but may be useful for development.
-
-By default, transformations are executed using Ray tasks.
-For transformations that require setup, specify ``compute=ray.data.ActorPoolStrategy(min, max)`` and Ray will use an autoscaling actor pool of ``min`` to ``max`` actors to execute your transforms.
-For a fixed-size actor pool, specify ``ActorPoolStrategy(n, n)``.
-The following is an end-to-end example of reading, transforming, and saving batch inference results using Ray Data:
-
-.. code-block:: python
-
-    from ray.data import ActorPoolStrategy
-
-    # Example of GPU batch inference on an ImageNet model.
-    def preprocess(image: bytes) -> bytes:
-        return image
-
-    class BatchInferModel:
-        def __init__(self):
-            self.model = ImageNetModel()
-        def __call__(self, batch: pd.DataFrame) -> pd.DataFrame:
-            return self.model(batch)
-
-    ds = ray.data.read_binary_files("s3://bucket/image-dir")
-
-    # Preprocess the data.
-    ds = ds.map(preprocess)
-    # -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1123.54it/s]
-
-    # Apply GPU batch inference with actors, and assign each actor a GPU using
-    # ``num_gpus=1`` (any Ray remote decorator argument can be used here).
-    ds = ds.map_batches(
-        BatchInferModel, compute=ActorPoolStrategy(10, 20),
-        batch_size=256, num_gpus=1)
-    # -> Map Progress (16 actors 4 pending): 100%|██████| 200/200 [00:07, 27.60it/s]
-
-    # Save the results.
-    ds.repartition(1).write_json("s3://bucket/inference-results")
+    Datasets also provides the convenience transformation methods :meth:`ds.map() <ray.data.Dataset.map>`,
+    :meth:`ds.flat_map() <ray.data.Dataset.flat_map>`, and :meth:`ds.filter() <ray.data.Dataset.filter>`,
+    which are not vectorized (slower than :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`), but
+    may be useful for development.
```

Comment:
@ericl Do we also want to call out `ds.add_column()`?

Reply:
Actually add_column is using map_batches.